pkg/auth/providers/github/github_client.go

```diff
@@ -118,11 +114,54 @@ func (g *GClient) getTeams(githubAccessToken string, config *v32.GithubConfig) (
 			logrus.Errorf("Github getGithubTeams: received error unmarshalling teams array, err: %v", err)
 			return teams, err
 		}
-		for _, teamObj := range teamObjs {
-			teams = append(teams, teamObj)
+		teams = append(teams, teamObjs...)
+
+	}
+	return teams, nil
+}
+
+// getOrgTeams returns the teams belonging to an organization.
+func (g *GClient) getOrgTeams(githubAccessToken string, config *v32.GithubConfig, org Account) ([]Account, error) {
+	url := fmt.Sprintf(g.getURL("ORG_TEAMS", config), org.Login)
```

Comment:
I think this should use [url.JoinPath](https://pkg.go.dev/net/url#JoinPath) ?

Reply:
Great observation. That was intentional as doing so requires to rethink or eliminate `getURL()`. As in https://github.com/rancher/rancher/pull/45162#discussion_r1571082863 the idea is to keep the scope of changes small. Using string interpolation here should be relatively safe as `org.Login` is not coming from the user. That being said we can always use `url.PathEscape()`.